pkg/cloudprovider/backup_service.go

```diff
@@ -145,12 +150,13 @@ func (br *backupService) UploadBackup(bucket, backupName string, metadata, backu
 	return nil
 }
 
-func (br *backupService) DownloadBackup(bucket, backupName string) (io.ReadCloser, error) {
-	return br.objectStore.GetObject(bucket, getBackupContentsKey(backupName))
+func (br *backupService) DownloadBackup(bucket, bucketPath, backupName string) (io.ReadCloser, error) {
+	return br.objectStore.GetObject(bucket, getBackupContentsKey(bucketPath, backupName))
 }
 
-func (br *backupService) GetAllBackups(bucket string) ([]*api.Backup, error) {
-	prefixes, err := br.objectStore.ListCommonPrefixes(bucket, "/")
+func (br *backupService) GetAllBackups(bucket string, bucketPath string) ([]*api.Backup, error) {
+	prefixes, err := br.objectStore.ListCommonPrefixes(bucket, "/", path.Clean(bucketPath)+"/")
```

Comment:
I had to modify this to the following to make it work when not specifying a path:

```
var prefix string
	if bucketPath != "" {
		prefix = path.Clean(bucketPath) + "/"
	}

	prefixes, err := br.objectStore.ListCommonPrefixes(bucket, "/", prefix)
```